File: Difficulty: Basic/Linked List End Insertion/linked-list-end-insertion.java
/*
class Node{
    int data;
    Node next;

    Node(int x){
        data = x;
        next = null;
    }
}
*/
class Solution {
    public Node insertAtEnd(Node head, int x) {
        Node newnode=new Node(x);
        Node curr=head;
        if(head==null){
            return newnode;
        }
        while(curr.next!=null){
            curr=curr.next;
            
        }
        curr.next=newnode;
        return head;
        
    }
}

File: Difficulty: Basic/Linked List End Insertion/linked-list-end-insertion.py
'''    
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None
'''

class Solution:
    def insertAtEnd(self, head, x):
        newnode=Node(x)
        curr=head
        if(head==None):
            return newnode
        while(curr.next!=None):
            curr=curr.next;
        curr.next=newnode
        return head
        
            
        
       
        